### DISCUSSION TITLE
Using jackson as jsonb provider

### DISCUSSION THREAD

**Discussion by nimo23** [quarkusio/quarkus#39184] [Q&A] [answered]
I have a bunch of `jsonb` annotated classes and want to switch from yasson to jackson. As jackson is also a `jsonb`-provider when using:

```
<dependency>
	<groupId>com.fasterxml.jackson.jakarta.rs</groupId>
	<artifactId>jackson-jakarta-rs-json-provider</artifactId>
</dependency>
```

it could be used in combination with

```
<dependency>
	<groupId>io.quarkus</groupId>
	<artifactId>quarkus-resteasy-reactive-jsonb</artifactId>
</dependency>
<dependency>
	<groupId>io.quarkus</groupId>
	<artifactId>quarkus-rest-client-reactive-jsonb</artifactId>
</dependency>
```
Is the above enough for Quarkus to register Jackson as a jsonb-provider? If I use Jackson as a jsonb-provider, I can also use Jackson annotations (e.g. `@JsonView`) in combination with jsonb-annotations. Using such an approach, will it fully work in Quarkus?

Another question is:

Would it be reasonable to switch the quarkus internal jsonb-provider to `jackson` instead of using `yasson`?

[a comment by an automated account is not shown]

**Comment by geoand** [comment]
> As jackson is also a jsonb-provider

This is **not** true

**Comment by nimo23** [comment]
>This is not true

@geoand According to https://github.com/FasterXML/jackson-jakarta-rs-providers and https://stackoverflow.com/questions/71645606/what-happens-if-we-have-both-jackson-jaxrs-json-provider-and-jersey-media-json-b, I came to the conclusion that with `jackson-jakarta-rs-json-provider` jackson can use jsonb-annotations for its processing, or did I miss something?

**Reply by geoand** [reply, marked as answer]
Jakarta REST (or JAX-RS as it used to be called) is **not** related to JSON-B, they are merely often used together. 

Jackson does **not** implement JSON-B

**Comment by nimo23** [comment]
Ok, then I need to convert jsonb annotations to jackson annotation. I was looking for a "Jsonb<-> Jackson"-annotation mapper and thought this lib would be the one. But unfortunately there is no such thing :=)

**Reply by geoand** [reply]
👍🏼
